src/lightning_lite/cli.py

```diff
@@ -0,0 +1,151 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from argparse import ArgumentParser, Namespace
+from typing import List, Tuple
+
+import torch.distributed.run as torchrun
+
+from lightning_lite.accelerators import CPUAccelerator, CUDAAccelerator, MPSAccelerator
+
+_SUPPORTED_ACCELERATORS = ("cpu", "gpu", "cuda", "mps", "tpu")
+_SUPPORTED_STRATEGIES = (None, "ddp", "dp", "deepspeed")
+_SUPPORTED_PRECISION = ("64", "32", "16", "bf16")
+
+
+def _parse_args() -> Tuple[Namespace, List[str]]:
+    parser = ArgumentParser(description="Launch your script with the Lightning Lite CLI.")
+    parser.add_argument("script", type=str, help="Path to the Python script with Lightning Lite inside.")
+    parser.add_argument(
+        "--accelerator",
+        type=str,
+        default="cpu",
+        choices=_SUPPORTED_ACCELERATORS,
+        help="The hardware accelerator to run on.",
+    )
+    parser.add_argument(
+        "--strategy",
+        type=str,
+        default=None,
+        choices=_SUPPORTED_STRATEGIES,
+        help="Strategy for how to run across multiple devices.",
+    )
+    parser.add_argument(
+        "--devices",
+        type=str,
+        default="1",
+        help=(
+            "Number of devices to run on (``int``), which devices to run on (``list`` or ``str``), or ``'auto'``."
+            " The value applies per node."
+        ),
+    )
+    parser.add_argument(
+        "--num-nodes",
+        type=int,
+        default=1,
+        help="Number of machines (nodes) for distributed execution.",
+    )
+    parser.add_argument(
+        "--node-rank",
+        type=int,
+        default=0,
+        help=(
+            "The index of the machine (node) this command gets started on. Must be a number in the range"
+            " 0, ..., num_nodes - 1."
+        ),
+    )
+    parser.add_argument(
+        "--main-address",
+        type=str,
+        default="127.0.0.1",
+        help="The hostname or IP address of the main machine (usually the one with node_rank = 0).",
+    )
+    parser.add_argument(
+        "--main-port",
+        type=int,
+        default=29400,
+        help="The main port to connect to the main machine.",
+    )
+    parser.add_argument(
+        "--precision",
+        type=str,
+        default="32",
+        choices=_SUPPORTED_PRECISION,
+        help=(
+            "Double precision (``64``), full precision (``32``), half precision (``16``) or bfloat16 precision"
+            " (``'bf16'``)"
+        ),
+    )
+
+    args, script_args = parser.parse_known_args()
+    return args, script_args
+
+
+def _set_env_variables(args: Namespace) -> None:
```

Comment:
Do you think these should be popped after the launched function finishes?

Reply:
sure, we can make it a context manager that restores the env on exit

Reply:
Any suggestion how to write the existing tests in test_cli in a way that is compatible with your suggestion? If the env variables don't survive after calling the cli, I'm not sure how to assert them.

Reply:
`assert "var" not in os.environ` should do the trick, right?

Reply:
That would be tautological if we delete the var after running the command. I want to test the opposite. See for example any test with name `test_cli_env_vars_*`.

Reply:
I think I will leave this as future work. I'm not concerned about this right now because the expected behavior is that the CLI exits when the called program exits. It won't invoke any other programs, so it is ok to leave the environ as it is.